packages/playwright/src/reporter.ts

```diff
@@ -175,25 +176,31 @@ export default class ReplayPlaywrightReporter implements Reporter {
     };
   }
 
-  onBegin({ version, projects }: FullConfig) {
-    const replayBrowserPath = getRuntimePath();
-    for (const project of projects) {
-      this._projects[project.name] = {
-        executed: false,
-        usingReplay: project.use.launchOptions?.executablePath === replayBrowserPath,
-      };
-    }
+  onBegin({ version }: FullConfig) {
     this.reporter.setTestRunnerVersion(version);
     this.reporter.onTestSuiteBegin();
   }
 
+  private _registerExecutedProject(test: TestCase) {
+    const project = test.parent.project();
+    if (project) {
+      let projectMetadata = this._executedProjects[project.name];
+      if (!projectMetadata) {
+        projectMetadata = this._executedProjects[project.name] = {
+          usesReplayBrowser: project.use.launchOptions?.executablePath === getRuntimePath(),
+        };
+      }
+      return projectMetadata;
+    }
+
+    return null;
+  }
+
   onTestBegin(test: TestCase, testResult: TestResult) {
-    const projectName = test.parent.project()?.name;
+    const projectMetadata = this._registerExecutedProject(test);
 
-    // it's important to handle the root project's name here and that's an empty string
-    if (typeof projectName === "string") {
-      this._projects[projectName].executed = true;
-    }
+    // Don't save metadata for non-Replay projects
+    if (projectMetadata?.usesReplayBrowser === false) return;
```

Comment:
It's unclear to me when it can happen but, in theory (aka according to the types), the project is optional.... In such a scenario the tested value resolves to `undefined` here and based on that code will continue to process this. I wonder if that's intentional behavior or not

Reply:
It was also unclear to me about when this could happen. (How would Playwright even know which browser to use if there was no project?)

I just did some local testing and realized that I can comment out the `projects` config block  entirely and Playwright still works:
```js
export default defineConfig({
  testDir: "./tests",
  fullyParallel: true,
  forbidOnly: !!process.env.CI,
  retries: 0,
  reporter: [
    ["dot"] as const,
    replayReporter({
      apiKey: process.env.REPLAY_API_KEY,
      upload: true,
    }),
  ],
  timeout: 10_000,
  use: {
    viewport: {
      width: 1280,
      height: 1024,
    },
  },
});
```
Adding some logging to the reporter I see that we _still_ get a project back in this case though:
```js
{
  grep: /.*/,
  grepInvert: null,
  outputDir: '/Users/bvaughn/Documents/git/replay/library/test-results',
  repeatEach: 1,
  retries: 0,
  metadata: {},
  name: '',
  testDir: '/Users/bvaughn/Documents/git/replay/library/tests',
  snapshotDir: '/Users/bvaughn/Documents/git/replay/library/tests',
  testIgnore: [],
  testMatch: '**/*.@(spec|test).?(c|m)[jt]s?(x)',
  timeout: 10000,
  use: { viewport: { width: 1280, height: 1024 } },
  dependencies: [],
  teardown: undefined,
  __projectId: ''
}
```

So the logic I have here would still hold. That being said, maybe it's safer to switch back to my original approach:
```js
if (!projectMetadata?.usesReplayBrowser) return;
```